lib/auth/auth.go

```diff
@@ -73,6 +73,12 @@ type ServerOption func(*Server)
 
 // NewServer creates and configures a new Server instance
 func NewServer(cfg *InitConfig, opts ...ServerOption) (*Server, error) {
+	err := utils.RegisterPrometheusCollectors(generateRequestsCount, generateThrottledRequestsCount,
```

Comment:
Nit: As a small stylistic suggestion, I would keep the list of prometheus collectors as a package-level variable, then this Register call would be cleaner when there are a lot of collectors and also easier to see all collectors:

```go
func NewServer(...) (*Server, error) {
  err := utils.RegisterPrometheusCollectors(prometheusCollectors)
  ...
}

var prometheusCollectors = []prometheusCollector{
  generateRequestCount,
  generateThrottleRequestsCount,
  ...
}
```